File: 11726 - 2xn 타일링/11726.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    cin.tie(0); ios::sync_with_stdio(0);
    int arr[1001] = {0, 1, 2};
    int n; cin >> n;
    for (int i = 3; i <= n; i++){
        arr[i] = (arr[i - 1] % 10007 + arr[i - 2] % 10007) % 10007;
    }
    cout << arr[n] % 10007 << '\n';
}

File: 11726 - 2xn 타일링/11726.py
arr = [0, 1, 2]
n = int(input())

for i in range(3, n + 1):
    arr.append((arr[i - 1] % 10007 + arr[i - 2] % 10007) % 10007)
print(arr[n] % 10007)
